### ISSUE TITLE
MapPattern with Int() key not working with compound value

### ISSUE THREAD

**Issue by justvanrossum** [crdoconnor/strictyaml#102]
I'm trying to write a schema using integer keys for a MapPattern, but am running into a snag:

```python
from strictyaml import MapPattern, Int, Str, load, Any

yaml = """
1:
- ABC
2:
- DEF
- GHI
"""

schema1 = MapPattern(Str(), Any())
schema2 = MapPattern(Int(), Any())

print(load(yaml).data)
print(load(yaml, schema1).data)
print(load(yaml, schema2).data)  # fails
```

Result:
```
OrderedDict([('1', ['ABC']), ('2', ['DEF', 'GHI'])])
OrderedDict([('1', ['ABC']), ('2', ['DEF', 'GHI'])])
Traceback (most recent call last):
  File "<untitled>", line 16, in <module>
  File ".../Python3.7/strictyaml/parser.py", line 318, in load
  File ".../Python3.7/strictyaml/parser.py", line 296, in generic_load
  File ".../Python3.7/strictyaml/validators.py", line 17, in __call__
  File ".../Python3.7/strictyaml/compound.py", line 75, in validate
  File ".../Python3.7/strictyaml/validators.py", line 17, in __call__
  File ".../Python3.7/strictyaml/any_validator.py", line 52, in validate
  File ".../Python3.7/strictyaml/validators.py", line 17, in __call__
  File ".../Python3.7/strictyaml/compound.py", line 257, in validate
  File ".../Python3.7/strictyaml/yamllocation.py", line 54, in process
  File ".../Python3.7/strictyaml/yamlpointer.py", line 179, in get
  File ".../Python3.7/ruamel/yaml/comments.py", line 753, in __getitem__
KeyError: '1'
```

Am I doing something wrong or are Int() keys not supported?

**Comment by crdoconnor** [comment]
No, that looks like a bug, and quite an embarrassing one too :(

I'm going to prioritize fixing this today.

**Comment by crdoconnor** [comment]
Hi, it's proving to be a little trickier than I thought to do this but I'm
still working on it. Thanks for your patience.

**Comment by justvanrossum** [comment]
(Thanks so much for looking into this. It's not a blocker for me, so no pressure from me.)

**Comment by tfuzeau** [comment]
Hi, I've just opened an issue before finding this one, I'm adding my note here as it makes more sense:

I think I found a bug with schema validation using a Map inside MapPattern with key as integer, please take a look to the following example I did:

```python
In [100]: data = """
     ...: vlans:
     ...:   1010:
     ...:     vrf: test
     ...:   1011:
     ...:     vrf: test
     ...:   1012:
     ...:     vrf: test
     ...: """
     ...:
     ...: schema = schema=Map({"vlans": MapPattern(Int(), Map({"vrf": Str()}))})
     ...: strictyaml.load(data, schema)
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-100-ff45bf5d6a8b> in <module>
     10
     11 schema = schema=Map({"vlans": MapPattern(Int(), Map({"vrf": Str()}))})
---> 12 strictyaml.load(data, schema)

~/Work/.venv/lib/python3.8/site-packages/strictyaml/parser.py in load(yaml_string, schema, label)
    316     and produce corresponding YAML object.
    317     """
--> 318     return generic_load(yaml_string, schema=schema, label=label)

~/Work/.venv/lib/python3.8/site-packages/strictyaml/parser.py in generic_load(yaml_string, schema, label, allow_flow_style)
    294         schema = Any()
    295
--> 296     return schema(YAMLChunk(document, label=label))
    297
    298

~/Work/.venv/lib/python3.8/site-packages/strictyaml/validators.py in __call__(self, chunk)
     15
     16     def __call__(self, chunk):
---> 17         self.validate(chunk)
     18         return YAML(chunk, validator=self)
     19

~/Work/.venv/lib/python3.8/site-packages/strictyaml/compound.py in validate(self, chunk)
    157                 )
    158
--> 159             value.process(self._validator_dict[yaml_key.scalar](value))
    160             key.process(yaml_key)
    161             chunk.add_key_association(key.contents, yaml_key.data)

~/Work/.venv/lib/python3.8/site-packages/strictyaml/validators.py in __call__(self, chunk)
     15
     16     def __call__(self, chunk):
---> 17         self.validate(chunk)
     18         return YAML(chunk, validator=self)
     19

~/Work/.venv/lib/python3.8/site-packages/strictyaml/compound.py in validate(self, chunk)
     73             yaml_key = self._key_validator(key)
     74             key.process(yaml_key)
---> 75             value.process(self._value_validator(value))
     76             chunk.add_key_association(key.contents, yaml_key.data)
     77

~/Work/.venv/lib/python3.8/site-packages/strictyaml/validators.py in __call__(self, chunk)
     15
     16     def __call__(self, chunk):
---> 17         self.validate(chunk)
     18         return YAML(chunk, validator=self)
     19

~/Work/.venv/lib/python3.8/site-packages/strictyaml/compound.py in validate(self, chunk)
    144     def validate(self, chunk):
    145         found_keys = set()
--> 146         items = chunk.expect_mapping()
    147
    148         for key, value in items:

~/Work/.venv/lib/python3.8/site-packages/strictyaml/yamllocation.py in expect_mapping(self)
    115             )
    116             for (regular_key, validated_key) in zip(
--> 117                 self.contents.keys(), self.strictparsed().keys()
    118             )
    119         ]

~/Work/.venv/lib/python3.8/site-packages/strictyaml/yamllocation.py in strictparsed(self)
    252
    253     def strictparsed(self):
--> 254         return self._pointer.get(self._strictparsed, strictdoc=True)

~/Work/.venv/lib/python3.8/site-packages/strictyaml/yamlpointer.py in get(self, document, strictdoc)
    177         for index_type, index in self._indices:
    178             if index_type == "val":
--> 179                 segment = segment[index[1] if strictdoc else index[0]]
    180             elif index_type == "index":
    181                 segment = segment[index]

~/Work/.venv/lib/python3.8/site-packages/ruamel/yaml/comments.py in __getitem__(self, key)
    751         # type: (Any) -> Any
    752         try:
--> 753             return ordereddict.__getitem__(self, key)
    754         except KeyError:
    755             for merged in getattr(self, merge_attrib, []):

KeyError: '1010'
```


```python
In [101]: strictyaml.load(data)                                                                                              
Out[101]: YAML(OrderedDict([('vlans', OrderedDict([('1010', OrderedDict([('vrf', 'test')])), ('1011', OrderedDict([('vrf', 'test')])), ('1012', OrderedDict([('vrf', 'test')]))]))]))
```

```python
In [102]: data = """
     ...: vlans:
     ...:   "1010":
     ...:     vrf: test
     ...:   "1011":
     ...:     vrf: test
     ...:   "1012":
     ...:     vrf: test
     ...: """
     ...:
     ...: schema = schema=Map({"vlans": MapPattern(Str(), Map({"vrf": Str()}))})
     ...: strictyaml.load(data, schema)
     ...:
Out[102]: YAML(OrderedDict([('vlans', OrderedDict([('1010', OrderedDict([('vrf', 'test')])), ('1011', OrderedDict([('vrf', 'test')])), ('1012', OrderedDict([('vrf', 'test')]))]))]))
```

```python
In [103]: data = """
     ...: vlans:
     ...:   1010: test
     ...:   1011: test
     ...:   1012: test
     ...: """
     ...:
     ...: schema = schema=Map({"vlans": MapPattern(Str(), Str())})
     ...: strictyaml.load(data, schema)
Out[103]: YAML(OrderedDict([('vlans', OrderedDict([('1010', 'test'), ('1011', 'test'), ('1012', 'test')]))]))
```

**Comment by crdoconnor** [comment]
Fixed in version 1.1.1. Thanks @justvanrossum and @tfuzeau !
